server/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -138,6 +139,7 @@ public class IndicesService extends AbstractLifecycleComponent
     private final ThreadPool threadPool;
     private final CircuitBreakerService circuitBreakerService;
     private final BigArrays bigArrays;
+    private final ClusterService clusterService;
```

Comment:
Please double check, this commit adds `clusterService` to `IndicesService`

Reply:
Good heads up! Can we inject the `ClusterService` to the `PeerRecoverySourceService` instead? This is the only usage and I see no reason to store it at the `IndicesService` only to provide it to some other component.

Or do you prefer it that way to avoid possible later conflicts?

Reply:
I can create an issue so we can revisit it later on, after we conclude the backports, wdyt?

Reply:
hm, I will try to apply what you suggest, I don't think it will be a big issue with further backports